### PR TITLE
NCN-10. POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by aclearskye**
In test: I removed seconds and milliseconds - accounting for time delay with test & db

model:
- checks if article is valid
- checks if user is valid
- inserts values into comments table

### CODE REVIEW

File: __tests__/article_id-comments-post.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    expect(body).toEqual(expected);
+  });
+  it("status:400, sending an incorrect type", async () => {
+    const { body } = await request(app)
+      .post("/api/articles/1/comments")
+      .send({ pokemon: "fire red" })
+      .expect(400);
+    const expected = { msg: "Bad Request!" };
+    expect(body).toEqual(expected);
+  });
```
Comment (scar1377) [inline __tests__/article_id-comments-post.test.js:55]: 404 article not found. /api/articles/9999/comments
400 no an id /api/articles/eatingBananas/comments
404 user not found .send({username: 'Tommy', body: "tests"})

File: model/article_id-comments-post-model.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  );
+
+  validateRes(validateIdQuery);
+
+  const userResult = await db.query(
+    "SELECT users.username FROM users WHERE users.username = $1;",
+    [posting.username]
+  );
+
+  if (!userResult.rows.length) {
```
Comment (scar1377) [inline model/article_id-comments-post-model.js:26]: if username is a string but doesn't exist in the database, then it's a 404 not found. But if it's not a string then it's a 400 bad request

**Review by scar1377 — COMMENTED** [review]
A couple of tests to add

File: __tests__/article_id-comments-post.test.js (hunk 1 above)
Comment (aclearskye) [reply __tests__/article_id-comments-post.test.js:55]: thank you for the feedback <3

File: __tests__/article_id-comments-post.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+const app = require("../app.js");
+const testData = require("../db/data/test-data");
+const seed = require("../db/seeds/seed.js");
+beforeEach(() => seed(testData));
+afterAll(() => {
+  db.end();
+});
+
+describe("201: POST - /api/articles/:article_id/comments", () => {
+  const posting = {
```
Comment (alexdevdavis) [inline __tests__/article_id-comments-post.test.js:12]: Best practice is to reserve verbs for function names

File: __tests__/article_id-comments-post.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    body: "I find this existance challenging",
+    comment_id: 19,
+    votes: 0,
+  };
+  it("status:201 - should send the request body an object with username and body", async () => {
+    const { body } = await request(app)
+      .post("/api/articles/1/comments")
+      .send(posting)
+      .expect(201);
+    console.log(body);
```
Comment (alexdevdavis) [inline __tests__/article_id-comments-post.test.js:28]: Remove floaty log

File: controller/article_id-comments-post-controller.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const {
+  postingArticleData,
+} = require("../model/article_id-comments-post-model");
+
+exports.postArticleData = async (req, res, next) => {
```
Comment (alexdevdavis) [inline controller/article_id-comments-post-controller.js:5]: This function name is a bit misleading / vague

File: model/article_id-comments-post-model.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+const db = require("../db/connection");
+const { validateRes, validateID } = require("./article_id-model");
+
+exports.postingArticleData = async (id, posting) => {
```
Comment (alexdevdavis) [inline model/article_id-comments-post-model.js:4]: Likewise re function name

**Review by alexdevdavis — APPROVED** [review]
Just a couple of things to address here, but you've done a good job :)

**Pull request merged**